network/src/protocols/disconnect_message.rs

```diff
@@ -63,3 +79,33 @@ impl ServiceProtocol for DisconnectMessageProtocol {
         });
     }
 }
+
+impl DisconnectMessageProtocol {
+    // parse disconnect message and return peer_ip, ban seconds and reason(if exists)
+    // refer to NetworkState::ban_session() for message format
+    fn parse_disconnect_message(
+        &self,
+        session_id: SessionId,
+        message: &str,
+    ) -> (String, String, String) {
+        let addr = self.0.with_peer_registry(|reg| {
+            reg.get_peer(session_id)
+                .filter(|peer| !peer.is_whitelist)
+                .map(|peer| peer.connected_addr.clone())
+                .map_or(String::default(), |maddr| maddr.to_string())
+        });
+
+        let mut ban_seconds = String::new();
+        if let Some(start) = message.find(char::is_numeric) {
```

Comment:
I don't think this code block is working for a ban message with any numeric char in the reason section, for example, with status code:
`Ban for 15 seconds, reason: HeadersIsInvalid(415)`

Reply:
alright, i will correct it.